urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/ia/algoritmosGeneticos/Estrategia.java

```diff
@@ -0,0 +1,435 @@
+			chromosomas[i] = new Chromosome(conf, genes);	
+			/**/int[] cromosoma = obtenerValoresCromosoma(chromosomas[i]);
+			/**///System.out.println(cromosoma[0]+","+cromosoma[1]+","+cromosoma[2]+","+cromosoma[3]);
+		}
+		Population population = new Population(conf,chromosomas);
+		return new Genotype(conf,population);
+	}
+	
+	public static int cantidadCompetidores(){
+		return leerPoblacion().length;
+	}
+	
+	private static int[] obtenerValoresCromosoma(IChromosome cromosoma) {
+		int[] valores = new int[4];
+		for (int i=0; i<4; i++){
+			valores[i]=getPesoDeGen(cromosoma, i);
+		}
+		return valores;
+	}
+	
+	private static int getPesoDeGen(IChromosome cromosoma, int i) {
+		Integer peso = (Integer) cromosoma.getGene(i).getAllele();
+		return peso;
+	}
+	
+	private static int[][] leerPoblacion() {
+		    int[][] poblacion = new int[17][5];  //tamańo población inicial -> 15
+			JFileChooser fileChooser = new JFileChooser();
+			//String nombreFichero = "poblacionInicial.txt";
+			String nombreFichero = "ultimaGeneracion.txt";
+			File fichero = new File(directorio.getPath(),nombreFichero);
+			fileChooser.setCurrentDirectory(fichero);		
+			try {
+				BufferedReader reader = new BufferedReader(new FileReader(fichero));
+				String linea = reader.readLine();
+				int i = 0;
+				while (linea != null) {
+					String[] dimension = linea.split ("-");
+					poblacion[i][4] = Integer.parseInt(dimension[0]);
+					dimension = dimension[1].split(",");
+					for (int j=0 ; j<dimension.length ; j++) {				
+						poblacion[i][j]=Integer.parseInt(dimension[j]);
+					}
+					linea = reader.readLine();
+					i++;
+				}
+				reader.close();
+			} catch (FileNotFoundException e1) {
+				e1.printStackTrace();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return poblacion;
+	}
+	
+	public static String poblacionToString(IChromosome[] cromosomas){
+		String poblacion = "";
+		String nombreFichero = "ultimaGeneracion.txt";
+		if (!directorio.exists()) {
+            directorio.mkdirs();
+        }
+		try {
+			FileWriter fichero = new FileWriter(directorio.getPath()+"/"+nombreFichero, false);
+			PrintWriter writer;
+			writer = new PrintWriter(fichero);
+			for (int i = 0; i < 16; i++) {
+				int[] cromosoma = obtenerValoresCromosoma(cromosomas[i]);
+				poblacion += numeroCromosomasCreados+"-"+cromosoma[0]+","+cromosoma[1]+","+cromosoma[2]+","+cromosoma[3];
+				if(i != cromosomas.length-1){
+					//poblacion += "\r\n";
+				}
+				numeroCromosomasCreados++;
+			}
+			writer.println(poblacion);
+			writer.close();
+		} catch (IOException e) {
+			System.out.println("No se ha podido escribir en el archivo torneo.txt");
+		}
+		return poblacion;
+	}
+	
+	public static void guardarDatosEvolucion(String datos, String nombreArchivo){
+		String nombreFichero = nombreArchivo;
+		if (!directorio.exists()) {
+            directorio.mkdirs();
+        }
+		try {
+			
+			boolean bandera = true;
+			if(nombreFichero == "parametrosIniciales.txt"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.